tests/calculations/test_train.py

```diff
@@ -0,0 +1,98 @@
+"""Tests for model train."""
+
+import pytest
+
+from aiida.common import InputValidationError, datastructures
+
+from aiida_mlip.data.config import JanusConfigfile
+
+
+def test_prepare_train(fixture_sandbox, generate_calc_job, janus_code, config_folder):
+    """Test generating singlepoint calculation job."""
+
+    entry_point_name = "janus.train"
+    config_path = config_folder / "mlip_train.yml"
+    config = JanusConfigfile(file=config_path)
+    inputs = {
+        "metadata": {"options": {"resources": {"num_machines": 1}}},
+        "code": janus_code,
+        "mlip_config": config,
+    }
+
+    calc_info = generate_calc_job(fixture_sandbox, entry_point_name, inputs)
+
+    retrieve_list = [
+        calc_info.uuid,
+        "aiida-stdout.txt",
+        "logs",
+        "results",
+        "checkpoints",
+        "test.model",
+        "test_compiled.model",
+    ]
+
+    print(sorted(calc_info.retrieve_list))
+    print(retrieve_list)
+    # Check the attributes of the returned `CalcInfo`
+    assert fixture_sandbox.get_content_list() == ["mlip_train.yml"]
+    assert isinstance(calc_info, datastructures.CalcInfo)
+    assert isinstance(calc_info.codes_info[0], datastructures.CodeInfo)
+    assert sorted(calc_info.retrieve_list) == sorted(retrieve_list)
+
+
+def test_file_error(fixture_sandbox, generate_calc_job, janus_code, config_folder):
+    """Test error if path for xyz is non existent."""
+
+    entry_point_name = "janus.train"
+    config_path = config_folder / "mlip_train.yml"
+
+    # Temporarily modify config file to introduce an error
+    with open(config_path, encoding="utf-8") as file:
+        right_path = file.read()
+
+    wrong_path = right_path.replace("mlip_train.xyz", "mlip_train_wrong.xyz")
+    with open(config_path, "w", encoding="utf-8") as file:
+        file.write(wrong_path)
```

Comment:
I would feel more comfortable if we weren't modifying a file in-place. This seems liable to fall over and leave a corrupted file in case of an abort. 

Suggest dumping to a temporary file instead. Possibly using `pytest`'s tmp_path with multiple files generated on the fly.

Reply:
fair enough, in general, would you say it' better to actually use a tmp_path or to just have different files saved and use those? because the reason I did this (and I would use a tmp_path for the same reason) is to not store many slightly different config files, so the folder looks tidier to me, but I could just do that, I don't know what way is considered better normally